src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -0,0 +1,173 @@
+package fs2.async.mutable
+
+import fs2._
+import fs2.async.AsyncExt.Change
+
+import fs2.async.{immutable, AsyncExt}
+
+
+
+/**
+ * Asynchronous queue interface. Operations are all nonblocking in their
+ * implementations, but may be 'semantically' blocking. For instance,
+ * a queue may have a bound on its size, in which case enqueuing may
+ * block until there is an offsetting dequeue.
+ */
+trait Queue[F[_],A] {
+
+
+  /**
+   * Enqueue one element in this `Queue`.
+   *
+   * Evaluates to `false` if the queue is full, indicating the `a` was not queued up.
+   * Evaluates to `true` if the `a` was queued up successfully.
+   *
+   * @param a `A` to enqueue
+   */
+  def enqueueOne(a: A): F[Boolean]
```

Comment:
I think we need to have two functions:

* `enqueue1(a: A): F[Unit]` - enqueues the `a`, or waits until the queue is empty
* `offerEnqueue1(a: A): F[Boolean]` - enqueues the `a` immediately and returns `true`, or returns `false` immediately if the queue is full

Both semantics are useful and you can't implement them in terms of each other.